### DISCUSSION TITLE
What is the shadow database needed for?

### DISCUSSION THREAD

**Discussion by Stadly** [langfuse/langfuse#1193] [Support] [answered]
The Prisma documentation states that a shadow database might be necessary when _developing_ in a cloud environment. It should not be necessary in production environments. How come it is then needed when self-hosting Langfuse in the cloud? I would have thought that was a production scenario that did not use `prisma migrate dev`?

https://www.prisma.io/docs/orm/prisma-migrate/understanding-prisma-migrate/shadow-database

**Comment by marcklingen** [comment, marked as answer]
Unless you run into issues, there is no need to set `SHADOW_DATABASE_URL`. For context, we deploy cloud.langfuse.com without using the shadow database.

The env was exposed based on this request: https://github.com/langfuse/langfuse/issues/738

prisma migrate dev is not used when deploying langfuse:
https://github.com/langfuse/langfuse/blob/33abe53acca0e30e53ccd1f0d9728f41e8e827db/entrypoint.sh#L21-L23

**Comment by Stadly** [comment]
I had to use `SHADOW_DATABASE_URL` when setting up Langfuse in Azure. I've had a look at the code base and could also not find any indication that `prisma migrate dev` or `prisma migrate diff` is used when deploying. Therefore I find it very strange that `SHADOW_DATABASE_URL` is needed, given this information in the [Prisma Docs](https://www.prisma.io/docs/orm/prisma-migrate/understanding-prisma-migrate/shadow-database):

![image](https://github.com/langfuse/langfuse/assets/7263579/e0ed7095-0a80-4219-b0a7-af37d67db015)

Anyway, it's not a big deal. I was just wondering :)
